pkg/trace/api/api.go

```diff
@@ -580,12 +580,19 @@ func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.
 	select {
 	// Wait for the semaphore to become available, allowing the handler to
 	// decode its payload.
-	// Afer the configured timeout, respond without ingesting the payload,
+	// After the configured timeout, respond without ingesting the payload,
 	// and sending the configured status.
 	case r.recvsem <- struct{}{}:
 	case <-time.After(time.Duration(r.conf.DecoderTimeout) * time.Millisecond):
+		log.Debugf("trace-agent is overwhelmed, a payload has been rejected")
 		// this payload can not be accepted
 		io.Copy(io.Discard, req.Body) //nolint:errcheck
+		switch v {
```

Comment:
additionally we must introduce new telemetry metric which will allow us to track these errors across orgs.

Reply:
not sure if you are referring to telemetry around rejected payloads, but this already exists a few lines below:

```go
r.tagStats(v, req.Header, "").PayloadRefused.Inc()
```

This results in the metric `datadog.trace_agent.receiver.payload_refused`, which is available in the OOTB Trace Agent dashboard, as well as accessible in the metrics explorer.